Backup and restore updates for 2.6

dockerymick opened the pull request:
https://github.com/stolostron/backlog/issues/25895

dockerymick committed:
Backup and restore updates for 2.6

dockerymick committed:
update the inline link

dockerymick committed:
Applying change 2 and multiple comments for improvements

dockerymick committed:
Small update

dockerymick committed:
Reordered the sections

dockerymick committed:
Adding other restore options

dockerymick committed:
couple more changes and suggestions added

dockerymick committed:
Moved comment

dockerymick committed:
Reorganizing content/modules

dockerymick committed:
more updates, adding samples

dockerymick committed:
Update backup_intro.adoc

Adding reference link at the end of the section

dockerymick committed:
Update backup_intro.adoc

fixed header level

dockerymick committed:
more updates

dockerymick committed:
Merge branch 'mj-backuprestore-25895' of github.com:open-cluster-management/rhacm-docs into mj-backuprestore-25895

dockerymick committed:
more updates

dockerymick committed:
more updates

dockerymick committed:
more updates

dockerymick committed:
more updates

dockerymick committed:
More changes

dockerymick committed:
made comment to update glossary, will create separate issue to address

dockerymick committed:
Update backup_intro.adoc

commits from live review

dockerymick committed:
Update manage_backup_restore.adoc

Addressed suggestion from issue https://github.com/stolostron/backlog/issues/25892

dockerymick commented on backup_restore/manage_backup_restore.adoc:
```diff
@@ -420,69 +384,198 @@ spec:
+When you restore a backup on a new hub cluster, make sure that the previous hub cluster, where the backup was created, is shut down. If that cluster is running, the previous hub cluster tries to reimport the managed clusters when the managed cluster reconciliation finds that the managed clusters are no longer available.
 
-All other passive hub clusters now restore passive data using the backup data created by the new primary hub cluster. _Hub N_ is now the primary hub cluster, managing clusters and backing up data.
+//Can you remind me of how the user shuts the cluster down? Do they simply run a command? Is there a 'shut down' button in the console that I might have missed?
```
@birsanv I missed this comment, this is the last one related to this

birsanv replied:
@dockerymick in a real disaster scenario, the primary hub is down, the user doesn't have to do that.
If the user want to test a disaster scenario and bring the primary hub down for that : either use the hive hibernate if the hub is a hive cluster or simply turn down the hub cluster virtual machines

dockerymick replied:
Thanks! @oafischer what are your thoughts on us adding a reference link or small instruction on how to disable your cluster. Is it beneficial to even add the reference?

oafischer replied:
I guess it depends on whether users often want to test a disaster scenario? I assume it's more of a niche so I wouldn't link. A note or small instruction is probably enough.

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
 = Backup and restore
 
-The cluster backup and restore operator provides disaster recovery solutions for {product-title} hub cluster failure. The cluster backup and restore operato runs on the hub cluster and depends on the link:https://github.com/openshift/oadp-operator[OADP Operator] to install Velero, and to create a connection from the hub cluster to the backup storage location where the data is stored. 
+The cluster backup and restore operator runs on the hub cluster and provides disaster recovery solutions for {product-title} hub cluster failures. When the hub cluster fails, although all managed clusters are still working fine, some features like alerting or cluster updates based on policy configurations, are not working properly anymore. Once the hub cluster is unavailable you need to have a re ...
```
That one sentence is still a little awkward. How about the following?

When the hub cluster fails, some features like policy configuration-based alerting or cluster updates stop working, even if all managed clusters still work fine.

birsanv replied:
@oafischer looks good!

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
 = Backup and restore
 
-The cluster backup and restore operator provides disaster recovery solutions for {product-title} hub cluster failure. The cluster backup and restore operato runs on the hub cluster and depends on the link:https://github.com/openshift/oadp-operator[OADP Operator] to install Velero, and to create a connection from the hub cluster to the backup storage location where the data is stored. 
+The cluster backup and restore operator runs on the hub cluster and provides disaster recovery solutions for {product-title} hub cluster failures. When the hub cluster fails, although all managed clusters are still working fine, some features like alerting or cluster updates based on policy configurations, are not working properly anymore. Once the hub cluster is unavailable you need to have a re ...
```
Few more streamlining suggestions:

Once the hub cluster is unavailable, you need a recovery plan to decide if recovery is possible, or if the data needs to be recovered from a newly deployed hub cluster.

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+[#active-passive-config]
+== Active passive configuration
+
+In an active passive configuration, there is one active hub cluster and passive hub clusters. An active hub cluster is also considered the primary hub cluster, which manages clusters and backs up resources at defined time intervals, using the `BackupSchedule.cluster.open-cluster-management.io` resource. 
```
Can we clarify the first part of line 27? Is there one passive hub cluster or multiple?

birsanv replied:
multiple

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+
+In an active passive configuration, there is one active hub cluster and passive hub clusters. An active hub cluster is also considered the primary hub cluster, which manages clusters and backs up resources at defined time intervals, using the `BackupSchedule.cluster.open-cluster-management.io` resource. 
+
+Passive hub clusters continuously retrieve the latest backups and restore the passive data. The passive hubs use the `Restore.cluster.open-cluster-management.io` resource to restore passive data from the primary hub cluster when new backup data is available. These hub clusters are on standby to become a primary hub when the primary hub cluster goes down.
```
Is `Restore.cluster.open-cluster-management.io` supposed to be capitalized?

dockerymick replied:
Yes, `Restore` is the resource kind. This syntax is the equivalent to a breadcrumb, but for YAMLs. However, we can just say "`Restore` resource". I think i went with the breadcrumb option bc I thought it would be repetitive. Let me know your thoughts @oafischer

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+
+Passive hub clusters continuously retrieve the latest backups and restore the passive data. The passive hubs use the `Restore.cluster.open-cluster-management.io` resource to restore passive data from the primary hub cluster when new backup data is available. These hub clusters are on standby to become a primary hub when the primary hub cluster goes down.
+
+Active and passive hub clusters are connected to the same storage location, where the primary hub cluster backs up data for passive hub clusters to access the primary hub cluster backups. For more details on how to setup this automatic restore configuration, see the <<restore-passive-resources-check-backups,Restoring passive resources while checking for backups>> section.
```
setup -> set up

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+[#disaster-recovery]
+== Disaster recovery
+
+When the primary hub cluster goes down, one of the passive hub clusters is chosen by the administrator to take over the managed clusters. In the following image, the administrator decides to use _Hub cluster N_ as the new primary hub cluster:
```
Can we rewrite "goes down"? Maybe shuts off? Or fails? Thinking about translation here.

oafischer replied:
I think we can switch to active voice here if the following still makes sense:

When the primary hub cluster shuts down, the administrator chooses a passive hub cluster to take over the managed clusters.

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+
+image:../images/disaster_recovery.png[Disaster recovery diagram] 
+
+_Hub cluster N_ restores the managed cluster activation data. At this point, the managed clusters connect with _Hub cluster N_. The administrator activates a backup on the new primary hub cluster, _Hub cluster N_, by creating a `BackupSchedule.cluster.open-cluster-management.io` resource, and storing the backups at the same storage location as the initial primary hub cluster.
```
Is `BackupSchedule.cluster.open-cluster-management.io` supposed to be capitalized? It's not capitalized in the next paragraph.

dockerymick replied:
Great catch, I'll make that update

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+
+The cluster backup and restore operator solution provides backup and restore support for all hub cluster resources like managed clusters, applications, policies, and bare metal assets. You can use the solution to back up any third-party resources extending the basic hub cluster installation. With this backup solution, you can define a cron-based backup schedule, which runs at specified time inter ...
+
+When the hub cluster needs to be replaced or is in a disaster scenario when the hub cluster goes down, a new hub cluster can be deployed and backed up data is moved to the new hub cluster.
```
Same comment about "goes down," is there something else we can use?

oafischer commented on backup_restore/backup_intro.adoc:
```diff
@@ -1,11 +1,191 @@
+    cluster.open-cluster-management.io/backup: ""
+----
++
+*Note:* Secrets used by the `hive.openshift.io.ClusterDeployment` resource need to be backed up, and are automatically annotated with the `cluster.open-cluster-management.io/backup` label only when the cluster is created using the console. If the Hive cluster is deployed using GitOps instead, the `cluster.open-cluster-management.io/backup` label must be manually added to the secrets used by the ` ...
```
Is Hive cluster capitalized? I've seen both in the docs.

dockerymick replied:
i usually capitalize bc that is how it's referenced in their docs. I only use lowercase if it is a pod name or api name i.e `clusterclaim.hive.openshift`

https://cwiki.apache.org/confluence/display/Hive//GettingStarted

oafischer reviewed (approved):
Left some comments but I'll approve ahead of time. Thanks so much for your work on this @dockerymick and @birsanv!

dockerymick committed:
Update backup_intro.adoc

Updates after peer review

dockerymick committed:
Update glossary_terms.adoc

Removed comment, created new issue: https://github.com/stolostron/backlog/issues/26417

dockerymick committed:
Update manage_backup_restore.adoc

Adding update from this issue: https://github.com/stolostron/backlog/issues/26288#issuecomment-1279409665

Related PR: https://github.com/stolostron/rhacm-docs/pull/4208

dockerymick committed:
Update manage_backup_restore.adoc

dockerymick committed:
Update backup_intro.adoc

fixing link

birsanv reviewed (approved):
lgtm

dockerymick commented:
@oafischer @birsanv Thanks so much for yall's approval! Merging this PR and making changes to the other versions
